tests/integrationtest/r/explain_shard_index.result

```diff
@@ -19,8 +19,8 @@ Projection	10.00	root		explain_shard_index.test3.id, explain_shard_index.test3.a
   └─TableRowIDScan(Probe)	10.00	cop[tikv]	table:test3	keep order:false, stats:pseudo
 explain format=brief select * from test3 where a=100 or b = 200;
 id	estRows	task	access object	operator info
-Projection	8000.00	root		explain_shard_index.test3.id, explain_shard_index.test3.a, explain_shard_index.test3.b
-└─Selection	8000.00	root		or(and(eq(tidb_shard(explain_shard_index.test3.a), 8), eq(explain_shard_index.test3.a, 100)), eq(explain_shard_index.test3.b, 200))
+Projection	11.00	root		explain_shard_index.test3.id, explain_shard_index.test3.a, explain_shard_index.test3.b
+└─Selection	10.01	root		or(and(eq(tidb_shard(explain_shard_index.test3.a), 8), eq(explain_shard_index.test3.a, 100)), eq(explain_shard_index.test3.b, 200))
```

Comment:
Are these estimation changes OK?

Reply:
The estimation change should be looked into. I missed this change.

Reply:
The most challenging problem of this pr is that the pseudo estimation logic is heavily influenced by this pr.
Many changes are to keep the pseudo-estimation behavior the same as before.